### ISSUE TITLE
Support CTCBeamSearchDecoder

### ISSUE THREAD

**Issue by CaptainDario** [onnx/tensorflow-onnx#2302]
# New Operator

### Describe the operator
Basic BeamSearch decoder as described [here](https://www.tensorflow.org/api_docs/python/tf/nn/ctc_beam_search_decoder)

### Do you know this operator be constructed using existing ONNX operators?
I think so but I am not sure

### Is this operator used by any model currently? Which one?
Many models use beam search for decoding the bare output such as CRNN.

### Are you willing to contribute it? (Y/N)
N

### Notes
I am trying to convert a CRNN model from TensorFlow that uses beam search.
I am fairly new to ONNX and I am getting this error when trying to convert it.

``` python
Tensorflow op [crnn/CTCBeamSearchDecoder: CTCBeamSearchDecoder] is not supported
Unsupported ops: Counter({'CTCBeamSearchDecoder': 1})
```

When looking at the [operator kernel list](https://github.com/microsoft/onnxruntime/blob/main/docs/OperatorKernels.md#dmlexecutionprovider) beam search is listed, therefore I assumed it would be supported.
Am I doing something wrong?

**Comment by fatcat-z** [comment]
Yes, it is highly possible that BeamSearch op in ORT could be mapped to CTCBeamSearchDecoder op directly. 

To make the conversion happen, we need to add a mapping and set the attributes/inputs accordingly. One example is [TensorScatterAdd](https://github.com/onnx/tensorflow-onnx/blob/358b519e14163c9087e778cfe8ff3b603ef88300/tf2onnx/onnx_opset/tensor.py#L664) op and adding relative tests.

And if possible, your contributions are definitely welcome!

**Comment by CaptainDario** [comment]
Thank you for your response!
I am interested in this but need to put down my research for the next months.
If nobody adds this maybe I will work on it in a few months but currently I am unsure.
